menoh/run_onnx.cpp

```diff
@@ -0,0 +1,458 @@
+#include <errno.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include <algorithm>
+#include <chrono>
+#include <cstdlib>
+#include <fstream>
+#include <map>
+#include <numeric>
+#include <queue>
+#include <set>
+#include <string>
+
+#include <common/log.h>
+#include <menoh/menoh.hpp>
+
+#ifdef _WIN32
+// HACK for Windows including order
+#define NOMINMAX
+#include <windows.h>
+#include <filesystem>
+namespace fs = std::experimental::filesystem;
+#undef OPAQUE
+#else
+#include <dirent.h>
+#include <unistd.h>
+#endif
+
+#include <chainerx/array.h>
+#include <chainerx/backprop_mode.h>
+#include <chainerx/context.h>
+#include <chainerx/native/native_backend.h>
+#include <chainerx/numeric.h>
+#include <chainerx/routines/creation.h>
+#include <chainerx/routines/manipulation.h>
+
+#include <common/protoutil.h>
+#include <common/strutil.h>
+#include <compiler/chxvm/emitter.h>
+#include <compiler/computation_order/core.h>
+#include <compiler/custom_onnx_ops.h>
+#include <compiler/flags.h>
+#include <compiler/flops.h>
+#include <compiler/gradient.h>
+#include <compiler/gradient_with_order.h>
+#include <compiler/graph.h>
+#include <compiler/model.h>
+#include <compiler/passes.h>
+#include <compiler/tensor.h>
+#include <compiler/util.h>
+#include <compiler/value.h>
+#include <runtime/chainerx_util.h>
+#include <runtime/chrome_tracing.h>
+#include <runtime/chxvm.h>
+#include <runtime/chxvm.pb.h>
+#include <runtime/chxvm_var.h>
+#include <runtime/meminfo.h>
+
+#include <tools/cmdline.h>
+#include <tools/util.h>
+
+const char* GREEN = "\033[92m";
+const char* RED = "\033[91m";
+const char* RESET = "\033[0m";
+
+bool g_quiet;
+
+#define LOG() \
+    if (!g_quiet) std::cerr
+
+namespace chainer_compiler {
+namespace runtime {
+namespace {
+
+bool IsDir(const std::string& filename) {
+    struct stat st;
+    CHECK_EQ(0, stat(filename.c_str(), &st)) << "failed to stat: " << filename << ": " << strerror(errno);
+    return S_IFDIR == (st.st_mode & S_IFMT);
+}
+
+chainerx::Array MakeArrayFromONNX(const onnx::TensorProto& xtensor) {
```

Comment:
I think it's probably better to add run_onnx_util.h or something to share the implementation of them.

Reply:
I see. I will modify it to share them so.